internal/nodes/dequeue_ingestion.go

```diff
@@ -470,26 +471,70 @@ func GetShardVSPrefix(key string) string {
 	return shardVsPrefix
 }
 
-func GetShardVSName(s string, key string) string {
+func GetShardVSName(s string, key string, shardSize uint32, prefix ...string) string {
 	var vsNum uint32
-	shardSize := lib.GetshardSize()
+	extraPrefix := strings.Join(prefix, "-")
+
 	if shardSize != 0 {
 		vsNum = utils.Bkt(s, shardSize)
 		utils.AviLog.Debugf("key: %s, msg: VS number: %v", key, vsNum)
 	} else {
 		utils.AviLog.Debugf("key: %s, msg: Processing dedicated VS", key)
 		//format: my-cluster--foo.com-dedicated for dedicated VS. This is to avoid any SNI naming conflicts
+		if extraPrefix != "" {
+			return lib.GetNamePrefix() + extraPrefix + "-" + s + "-dedicated"
+		}
 		return lib.GetNamePrefix() + s + "-dedicated"
 	}
 	shardVsPrefix := GetShardVSPrefix(key)
+	if extraPrefix != "" {
+		shardVsPrefix += extraPrefix + "-"
+	}
 	vsName := shardVsPrefix + fmt.Sprint(vsNum)
-	utils.AviLog.Infof("key: %s, msg: ShardVSName: %s", key, vsName)
 	return vsName
 }
 
-func DeriveShardVS(hostname string, key string) string {
-	// Read the value of the num_shards from the environment variable.
+func AviInfraSettingChange(routeIgrObj RouteIngressModel) (*akov1alpha1.AviInfraSetting, *akov1alpha1.AviInfraSetting) {
+	var oldAviInfraSetting *akov1alpha1.AviInfraSetting
+	var err error
+	if found, oldInfraSettingName := objects.InfraSettingL7Lister().GetIngRouteToInfraSetting(routeIgrObj.GetNamespace() + "/" + routeIgrObj.GetName()); found {
+		oldAviInfraSetting, err = lib.GetCRDInformers().AviInfraSettingInformer.Lister().Get(oldInfraSettingName)
+		if err != nil {
+			utils.AviLog.Warnf("AviInfraSetting not found %s", err.Error())
+		}
+	}
+
+	aviInfraSetting := routeIgrObj.GetAviInfraSetting()
+	return oldAviInfraSetting, aviInfraSetting
+}
+
+// returns old and new models if changed, else just the current one.
+func DeriveShardVS(hostname string, key string, routeIgrObj RouteIngressModel) (string, string) {
+	// Read the value of the num_shards from the environment variable or the AviInfraSetting CRD.
```

Comment:
this comment says that we are reading the num_shards value, but I don't see it here...so a bit confused.